graphql_compiler/tests/snapshot_tests/test_query_pagination.py

```diff
@@ -513,6 +513,43 @@ def test_parameter_value_generation_int(self):
         expected_parameters = [26, 51, 76]
         self.assertEqual(expected_parameters, list(generated_parameters))
 
+    @pytest.mark.usefixtures("snapshot_orientdb_client")
+    def test_parameter_value_generation_int_few_quantiles(self):
+        schema_graph = generate_schema_graph(self.orientdb_client)
+        graphql_schema, type_equivalence_hints = get_graphql_schema_from_schema_graph(schema_graph)
+        pagination_keys = {vertex_name: "uuid" for vertex_name in schema_graph.vertex_class_names}
+        pagination_keys["Species"] = "limbs"  # Force pagination on int field
+        uuid4_fields = {vertex_name: {"uuid"} for vertex_name in schema_graph.vertex_class_names}
+        class_counts = {"Species": 10000000}
+        statistics = LocalStatistics(
+            class_counts, field_quantiles={("Species", "limbs"): [
+                0, 10, 20, 30,
+            ],}
+        )
+        schema_info = QueryPlanningSchemaInfo(
+            schema=graphql_schema,
+            type_equivalence_hints=type_equivalence_hints,
+            schema_graph=schema_graph,
+            statistics=statistics,
+            pagination_keys=pagination_keys,
+            uuid4_fields=uuid4_fields,
+        )
+
+        query = """{
+            Species {
+                name @output(out_name: "species_name")
+            }
+        }"""
+        args = {}
+        query_ast = safe_parse_graphql(query)
+        vertex_partition = VertexPartitionPlan(("Species",), "limbs", 3)
+        generated_parameters = generate_parameters_for_vertex_partition(
+            schema_info, query_ast, args, vertex_partition
+        )
+
+        expected_parameters = [26, 51, 76]
```

Comment:
I don't understand this bit. Hopefully you can explain when you put the fix together.

Reply:
It's incorrect, but the code has an `IndexError` anyway